src/Gui/View3DPy.cpp

```diff
@@ -777,10 +777,8 @@ Py::Object View3DInventorPy::getCameraOrientation()
 
 Py::Object View3DInventorPy::viewPosition(const Py::Tuple& args)
 {
-    PyObject* p=nullptr;
-    int steps = 20;
-    int ms = 30;
-    if (!PyArg_ParseTuple(args.ptr(), "|O!ii",&Base::PlacementPy::Type,&p,&steps,&ms))
+    PyObject* p = nullptr;

```

Comment:
This change is problematic because it may break existing Python macros.

Reply:
Yes true, we could either keep the steps and ms parameters and ignore them so it does not break, or we could break it. I feel like it is better to clean it up and make any breaking changes before a 1.0 release. But let me know what you think.

Reply:
In the past it was possible to write this Python code:
```
view = Gui.ActiveDocument.ActiveView

# Manually move to the desired position and then call...
pos1 = view.viewPosition()
# Manually move to the desired position and then call...
pos2 = view.viewPosition()
# Manually move to the desired position and then call...
pos3 = view.viewPosition()

view.viewPosition(pos1, 20, 2000)
view.viewPosition(pos2, 50, 5000)
view.viewPosition(pos3, 10, 1000)
```

To achieve a similar effect now the code must be changed to:
```
view = Gui.ActiveDocument.ActiveView

# Manually move to the desired position and then call...
pos1 = view.viewPosition()
# Manually move to the desired position and then call...
pos2 = view.viewPosition()
# Manually move to the desired position and then call...
pos3 = view.viewPosition()

par = App.ParamGet("User parameter:BaseApp/Preferences/View")
par.SetInt("AnimationDuration", 2000)
view.viewPosition(pos1)
par.SetInt("AnimationDuration", 5000)
view.viewPosition(pos2)
par.SetInt("AnimationDuration", 1000)
view.viewPosition(pos3)
```
The downside is that this has side-effects because changing AnmimationDuration parameter affects other functions. Therefore I recommend the following changes:
* Revert this change
* Add an overloaded method `View3DInventorViewer::moveCameraTo(const SbRotation& orientation, const SbVec3f& position, int duration)`
* Use this function here and pass "ms" as third parameter.
* "steps" will still be used for the Python interface but not used any more
* In startAnimation() replace the bool with an int and remove the access to the user parameters

Reply:
Sounds like a good plan, working on it.

Reply:
I added a commit that makes `view.viewPosition()` work as previously. If the duration is not passed to `viewPosition()` then it will be -1 by default. I did not overload `View3DInventorViewer::moveCameraTo(...)`  but instead I added a duration parameter with default value -1. `View3DInventorViewer::startAnimation(...)` now also has a duration parameter with default value -1. `View3DInventorViewer::startAnimation(...)` checks if the duration is valid, meaning more or equal to 0. If it is not valid then the duration is overwritten with the "AnimationDuration" parameter. So, If the duration is not passed explicitly in one of the previously mentioned methods then the duration will default to "AnimationDuration".

I also did not replace the bool "wait" because for `viewPosition()` it is desired to wait until the animation is finished but when selecting a standard view or a face on the navicube, waiting cause some side-effects (I don´t remember what exactly).

Reply:
> I also did not replace the bool "wait" because for viewPosition() it is desired to wait until the animation is finished

OK.